htdocs/core/js/lib_foot.js.php

```diff
@@ -196,3 +196,7 @@
 					}
 				});
 			});'."\n";
+// hide topmenus disabled
+if (!empty($conf->global->MAIN_BUTTON_HIDE_UNAUTHORIZED) && (!$user->admin)) {
```

Comment:
If option MAIN_BUTTON_HIDE_UNAUTHORIZED, hidden of menu should already be done by the menu manager into code core/menus/standard/eldy.lib.php
Don't you have this ?